### ISSUE TITLE
How to test routed components and shallow rendering?

### ISSUE THREAD

**Issue by checketts** [getsaf/shallow-render#28]
I apologize if this is too basic of a question, but I couldn't seem to find it in the docs. I'm liking the `shallow-render` approach, however several of my components are route destinations. Is there any hook I should use for testing those?

Or should I stick with the router testing from the Angular site? (https://angular.io/guide/testing#routed-components)

**Comment by checketts** [comment]
Thanks for the great example! I got it working (had to import the correct `Location` in my own code). 
 I hadn't realized the template snippet in `shallow.render()`had been optional, so that moved me closer. 

However, what I'm particularly interested in is testing what components get rendered based on a router. (Though the navigation example you provided is definitely another usecase I'm investigating) 


Here is my test case:
1- Navigate by URL
2- Look up whatever component got instantiated and what params got parsed out and make assertions on it.

```
    const router = get(Router);
    await router.navigateByUrl('/go');

    //look up the currently loaded component is
    const currentComponent = router.getCurrentComponent; //Any idea where this might be?
    expect(currentComponent).isA(GoHomeComponent) //psuedo assertion, I'll find the proper one.
```

**Comment by getsaf** [comment]
Gotcha, you are basically testing your routes, not your components. That *may* be outside of the scope of Shallow at the moment. The concept is really focused on testing components and mocking everything else. I do plan on expanding functionality a bit to possibly support services and other injectables, but I need a bit more experience with the Angular router to see if that fits the model too.

I noticed in the [link you gave](https://angular.io/guide/testing#routed-components) in the first post they don't really cover your scenario either. They do cover stubbing injected router objects into components for the purpose of testing your component (which we do support in Shallow). It almost seems like testing "when navigating to X ensure Y is rendered" is just writing unit tests for Angular Router or testing your configuration of Angular Router, neither of which seem to have much value when testing your own code (unless you consider testing configuration valuable). I stumbled around in StackBlitz for a while trying to figure out how you might do this but it seems Angular doesn't expose anything that allows you go get the currently-routed-component instance.

Sorry for the rant, just thinking out loud in case it helps.

TL;DR; I don't think Shallow can help with route configuration testing and I'm not aware of any patterns or tools that will either.

Sorry I wasn't much help here. Let me know if you figure this out.

**Comment by checketts** [comment]
I came to the same conclusion. I'm used to testing backend endpoints so my mindset aligns with: hit `/api/users` and verify it returns what is expected.

You actually were tons of help. Your example showed me how to render the component and that the snippet was optional and also to use `get` to find injectables.  If the router had exposed the info I needed, then it would have been perfect.

The reason I wanted to check if Shallow supported my usecase was because if the router was to return my component, I wanted it to be a shallow/mocked out version. For now, I'll pause on testing routes and stick with the common component tests instead.

Thanks for the help!
